tests/v2/validation/provisioning/airgap/k3s_custom_cluster_test.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/rancher/rancher/tests/framework/extensions/provisioning/verify"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2/validation/pipeline/rancherha/corralha"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning/permutations"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning/registries"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type AirGapK3SCustomClusterTestSuite struct {
+	suite.Suite
+	client         *rancher.Client
+	session        *session.Session
+	corralPackage  *corral.CorralPackages
+	clustersConfig *provisioning.Config
+	registryFQDN   string
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) TearDownSuite() {
+	a.session.Cleanup()
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	a.session = testSession
+
+	a.clustersConfig = new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, a.clustersConfig)
+
+	corralRancherHA := new(corralha.CorralRancherHA)
+	config.LoadConfig(corralha.CorralRancherHAConfigConfigurationFileKey, corralRancherHA)
+
+	registriesConfig := new(registries.Registries)
+	config.LoadConfig(registries.RegistriesConfigKey, registriesConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(a.T(), err)
+
+	a.client = client
+	listOfCorrals, err := corral.ListCorral()
+	require.NoError(a.T(), err)
+
+	corralConfig := corral.CorralConfigurations()
+
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(a.T(), err)
+
+	a.corralPackage = corral.CorralPackagesConfig()
+
+	_, corralExist := listOfCorrals[corralRancherHA.Name]
+	if corralExist {
+		bastionIP, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryIP)
+		require.NoError(a.T(), err)
+
+		err = corral.UpdateCorralConfig(corralBastionIP, bastionIP)
+		require.NoError(a.T(), err)
+
+		registryFQDN, err := corral.GetCorralEnvVar(corralRancherHA.Name, corralRegistryFQDN)
+		require.NoError(a.T(), err)
+		logrus.Infof("registry fqdn is %s", registryFQDN)
+		a.registryFQDN = registryFQDN
+	} else {
+		a.registryFQDN = registriesConfig.ExistingNoAuthRegistryURL
+	}
+}
+
+func (a *AirGapK3SCustomClusterTestSuite) TestProvisioningK3SCustomCluster() {
+	a.clustersConfig.NodesAndRoles = []machinepools.NodeRoles{provisioning.AllRolesPool}
+
+	tests := []struct {
+		name   string
+		client *rancher.Client
+	}{
+		{provisioning.AdminClientName.String() + "-" + permutations.K3SAirgapCluster + "-", a.client},
+	}
+	for _, tt := range tests {
+		a.Run(tt.name, func() {
+			testConfig := permutations.ConvertConfigToClusterConfig(a.clustersConfig)
+			clusterObject, err := crud.CreateProvisioningAirgapCustomCluster(tt.client, testConfig, a.corralPackage)
```

Comment:
There was a misunderstanding here Caleb this test runs multiple k8s versions, so perfect to be used with the RunTestPermutations.